pkg/ocm/machines.go

```diff
@@ -224,14 +225,17 @@ func (mtl *MachineTypeList) UpdateAvailableQuota(quotaCosts *amsv1.QuotaCostList
 		quotaCosts.Each(func(quotaCost *amsv1.QuotaCost) bool {
 			for _, relatedResource := range quotaCost.RelatedResources() {
 				if machineType.MachineType.GenericName() == relatedResource.ResourceName() && isCompatible(relatedResource) {
-					availableQuota := (quotaCost.Allowed() - quotaCost.Consumed()) / relatedResource.Cost()
-					machineType.Available = availableQuota > 1
-					machineType.availableQuota = availableQuota
+					if relatedResource.Cost() == 0 {
+						machineType.nodeQuota = math.Inf(+1)
+					} else {
+						machineType.nodeQuota += float64((quotaCost.Allowed() - quotaCost.Consumed()) / relatedResource.Cost())
+					}
```

Comment:
Wouldn't it be better to avoid the float math and keep it as ints? since we can't have 1.4 nodes anyways, and i'm worried about getting something like 1.999999999 due to quirky float arithmetic edge cases.
```suggestion
					if relatedResource.Cost() == 0 {
						machineType.nodeQuota = math.MaxInt
					} else {
						machineType.nodeQuota += (quotaCost.Allowed() - quotaCost.Consumed()) / relatedResource.Cost()
					}
```

Reply:
Note that the division itself remained int/int, rounding down; it's cast to a float afterwards.

I'm using floats for their ability to represent infinity, which fits checks like `nodeQuota > getDefaultNodes(multiAZ)` very naturally.  
(I've used a similar +Inf representation widely in the UI and consider it a success.)

Hmm yes, `MaxInt` could play a similar role in practice.  I find it less "elegant" but it's workable.  Do you want me to change it?